packages/x-date-pickers/src/PickersActionBar/PickersActionBar.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import DialogActions from '@mui/material/DialogActions';
+import { styled } from '@mui/material/styles';
+import { useLocaleText } from '../internals/hooks/useUtils';
+
+export type PickersActionBarAction = 'clear' | 'cancel' | 'accept' | 'today';
+
+export interface PickersActionBarProps {
+  /**
+   * Ordered array of actions to display.
+   * If empty, does not display that action bar.
+   * @default `['cancel', 'accept']` for mobile and `[]` for desktop
+   */
+  actions?: PickersActionBarAction[];
+  onAccept: () => void;
+  onClear: () => void;
+  onCancel: () => void;
+  onSetToday: () => void;
+}
+
+const PickersActionBarRoot = styled(DialogActions)<{
+  ownerState: PickersActionBarProps;
+}>(({ ownerState }) => ({
+  ...((ownerState.actions?.includes('clear') || ownerState.actions?.includes('today')) && {
```

Comment:
Why do we only add this CSS override for some actions ?

Reply:
To have the same behavior as the `<PickersModalDialogActions />` I'm replacing

```jsx
const PickersModalDialogActions = styled(DialogActions)<{
  ownerState: PickersModalDialogProps;
}>(({ ownerState }) => ({
  ...((ownerState.clearable || ownerState.showTodayButton) && {
    // set justifyContent to default value to fix IE11 layout bug
    // see https://github.com/mui-org/material-ui-pickers/pull/267
    justifyContent: 'flex-start',
    '& > *:first-of-type': {
      marginRight: 'auto',
    },
  }),
}));
```

Makes me realize that desktop has a different behavior

```jsx
const PickersPopperAction = styled(DialogActions)<{
  ownerState: PickerPopperProps;
}>(({ ownerState }) => ({
  ...(ownerState.clearable
    ? {
        justifyContent: 'flex-start',
        '& > *:first-of-type': {
          marginRight: 'auto',
        },
      }
    : { padding: 0 }),
}));
```

And the `marginRight: 'auto'` does not have any more sens now that users can modify buttons order.

I think I missed a big blocker: the difference between mobile and desktop which have different style and behavior

https://codesandbox.io/s/basicdatepicker-material-demo-forked-8gvp4h?file=/demo.tsx

So I propose to add 3 props to this component: 
- `isDesktop` such that developers can use `style(PickersActionBar)` to apply different style on descktop/mobile
- add `actionsForMobile` and `actionsForDesktop` by default to `undefined` if the dev specify them it overrides the `actions` props

Reply:
You can use `WrapperVariantContext` instead of adding a `isDesktop` prop.

> add actionsForMobile and actionsForDesktop by default to undefined if the dev specify them it overrides the actions props

I propose to rather implement https://github.com/mui/material-ui/issues/32144 in a follow up PR with the variant as a param.

Reply:
I added the callback for `actions` + some tests